app/src/main/java/com/ggetters/app/data/model/supers/CodedEntity.kt

```
@@ -61,13 +61,32 @@ interface CodedEntity {
+    fun generateCode() {
+        if (this is KeyedEntity) {
+            // Only generate if code is not already set to preserve idempotency
+            if (code == null) {
+                code = generateAlphanumericCode(6)
+            }
+        } else {
+            Clogger.e(
+                TAG, "Unique code can only be generated for valid database entities."
+            )
+            
+            throw IllegalStateException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)